backend/uchicago/watcher.py

```diff
@@ -0,0 +1,67 @@
+import requests
+from lxml import html
+import pyrebase
+from difflib import Differ
+import sendgrid
+import os
+from sendgrid.helpers.mail import *
+
+FIREBASE = pyrebase.initialize_app({
+    'apiKey':
+    'AIzaSyCjBDyhwbXcp9kEIA2pMHLDGxmCM4Sn6Eg',
+    'authDomain':
+    'canigraduate-43286.firebaseapp.com',
+    'databaseURL':
+    'https://canigraduate-43286.firebaseio.com',
+    'storageBucket':
+    'canigraduate-43286.appspot.com',
+    'serviceAccount':
+    './service_account_key.json'
+})
+
+def watcher(program):
+	url = db.child('programs').child(program).child('metadata').child('catalog').get().val()
+	xpath = db.child('programs').child(program).child('metadata').child('source').get().val()
+	stored_requirement = db.child('programs').child(program).child('page-state').get().val()
+	tree = html.fromstring(requests.get(url).content)
+	current_requirement = tree.xpath(xpath)[0].text_content()
+	if stored_requirement == None:
+		db.child('programs').child(program).child('page-state').set(current_requirement)
+	elif stored_requirement != current_requirement:
+		db.child('programs').child(program).child('page-state').set(current_requirement)
+		emailHelper(program, stored_requirement, current_requirement)
+
+	extensions = db.child('programs').child(program).child('extensions').get().val()
+	if extensions != None:
+		for e in list(extensions):
+			e_url = db.child('programs').child(program).child('extensions').child(e).child('metadata').child('catalog').get().val()
```

Comment:
Can you just call `watcher(extension)`?

Reply:
No, because `extension` is a child of its parent program, not `db.child('programs')`. But if we change the way major specialization programs are stored in db, I'd be able to. I was hoping to have a helper that takes in a path to the most immediate parent, but I ran into that strange problem.

What do you think?

Reply:
Ah, this is okay for now then. I think storing extensions should be in the scope of the parent, it'll make things easier to render. We should probably look into why the db path doesn't persist and fix it in the future. Can you ask on stackoverflow or file a bug with Pyrebase?

Reply:
Sounds good. I will do the two you suggested.